### PR TITLE
Add custom DNS resolver

### PR DESCRIPTION
**Pull request by CorentinB**

### CODE REVIEW

File: dns.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+		cachedEntry := cached.(cachedIP)
+		if time.Now().Before(cachedEntry.expiresAt) {
+			return resolvedIP, nil
+		}
+		// Cache entry expired, remove it
+		d.DNSRecords.Delete(address)
+	}
+
+	m := new(dns.Msg)
+	m.SetQuestion(dns.Fqdn(address), dns.TypeA)
```
Comment (equals215) [inline dns.go:40]: `dns.TypeAAAA` should be asked too

Comment (equals215) [reply dns.go:40]: but not in the same request, see : https://github.com/miekg/dns/issues/1092

**Review by equals215 — CHANGES_REQUESTED** [review]
other than archiving both A and AAAA records, all the rest looks good!

File: dns.go (hunk 1 above)
Comment (CorentinB) [reply dns.go]: Done! Thank you

**Pull request merged**